prac_08/taxi_simulator.py

```diff
@@ -0,0 +1,97 @@
+
+from prac_08.taxi import Taxi
+from prac_08.silver_service_taxi import SilverServiceTaxi
+
+MENU = "q)uit, c)hoose taxi, d)rive"
+
+
+def main():
+
+    total_bill = 0
+    current_taxi = ""
+    taxis = [Taxi("Prius", 100), SilverServiceTaxi("Limo", 100, 2),
+             SilverServiceTaxi("Hummer", 200, 4)]
+
+    print("Let's drive!")
+    print(MENU)
+    menu_selection = input(">>> ").lower()
+
+    while menu_selection != "q":
+        if menu_selection == "c":
+            print("Taxis Available: ")
+            list_available_taxis(taxis)
+            taxi_choice = receive_taxi_choice(taxis)
+            current_taxi = taxis[taxi_choice]
+            print("Bill to date: ${:.2f}".format(total_bill))
+            print(MENU)
+            menu_selection = input(">>> ").lower()
+        elif menu_selection == "d":
+            if current_taxi != "":
+                current_taxi.start_fare()
+                distance_to_drive = receive_distance_to_drive()
+                current_taxi.drive(distance_to_drive)
+                trip_cost = current_taxi.get_fare()
+                print("Your {} trip cost you ${:.2f}".format(current_taxi.name, trip_cost))
+                total_bill += trip_cost
+                print("Bill to date: ${:.2f}".format(total_bill))
+                print(MENU)
+                menu_selection = input(">>> ").lower()
+            else:
+                print("Please choose a taxi from the list before driving.")
+        else:
+            print("Invalid menu input")
+            print("Bill to date: ${:.2f}".format(total_bill))
+            print(MENU)
```

Comment:
You can pull the menu and choice out so it will run after each program is complete since all but q return to the menu